lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name,
+                :address,
+                :phone,
+                :services,
+                :registered_vehicles,
+                :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    @services.include?('Vehicle Registration')
+      collect_fees(vehicle)
+      vehicle.registration_date = Date.today
+      vehicle.set_plate_type
+      @registered_vehicles << vehicle
+  end
+
+  def collect_fees(vehicle)
+    if vehicle.antique?
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end
+  end
+
+  def administer_written_test(registrant)
+    if registrant.permit? == true && @services.include?('Written Test') && registrant.age >= 16
```

Comment:
In an conditional statement, you actually don't need to say `== true` for a boolean variable. If you just did:
`if registrant.permit?` alone, that will evaluate to true or false and serve the same purpose. It can clean up these lines. 

Also, it's generally not super readable to have a conditional with 3 different conditions X && Y && Z. Consider breaking those up into nested if statements. Sometimes it can be tempting to make everyone one line even if it's not very readable